app/assets/javascripts/date_selector.js

```diff
@@ -0,0 +1,72 @@
+var dateSelector = (function() {
+  var defaultOptions = {
+    mediatedPageFieldSelector: 'input[data-request-type="mediated_page"]'
+  };
+
+  return {
+    init: function(opts) {
+      var _this = this;
+      _this.options = $.extend(defaultOptions, opts);
+      $(document).on('ready page:load', function(){
+        _this.setupListeners();
+      });
+    },
+
+    options: {},
+
+    setupListeners: function() {
+      var _this = this;
+      if ( _this.mediatedDateField().length > 0 ) {
+        pagingScheduleUpdater
+          .schedulerText()
+          .on('paging-schedule:updated', function(_, data) {
+            _this.setMinDate(data.date);
+            _this.restrictToOpenDates(
+              data.destination_business_days
```

Comment:
This comes from a json response (that we do control...) but seems like this is unimportant.